### DISCUSSION TITLE
Refresh token on SSR not working.

### DISCUSSION THREAD

**Discussion by bnjtgly** [nuxt-community/auth-module#1552] [Support Requests]
Hi, does anyone knows how to make refresh token work? I'm using an API. Here's my nuxt.config.js. Everytime I added the refresh scheme my login stops working. Thank you so much.

```
auth: {
  strategies: {
    local: {
      token: {
        property: 'token',
          global: true
      },
      user: {
        property: 'user'
      },
      endpoints: {
        login: { url: 'login', method: 'post' },
        logout: { url: 'logout', method: 'delete' },
        user: { url: 'admin_api/profile', method: 'get' }
      }
    }
  },
  router: {
    middleware: ['auth']
  }
}
```

login.vue
`await this.$auth.loginWith('local', { data: { user: this.credentials } })`

**Comment by bnjtgly** [comment]
**UPDATE:**

Hi, I was able to make it work using refresh scheme but still encountered another error. Is there a way to include the expired token in the request? Thank you.
![image](https://user-images.githubusercontent.com/84297158/138007128-6dc3cffc-480d-4276-ba56-d24fe43ccaea.png)

**Comment by sts-ryan-holton** [comment]
@bnjtgly Ive got a similar issue, I'm using the Laravel JWT scheme to refresh tokens but in my network I keep seeing about 7 hits to my refresh endpoint in quick succession, the first succeeds, then the other fail and the app logs me out, don't know why.

Linked [issue](https://github.com/nuxt-community/auth-module/issues/1329)

**Comment by williansebastiao** [comment]
i'm having the same problem.
Both token and refresh token do not work and my page is redirected to login

![Captura de tela de 2021-10-21 11-43-35](https://user-images.githubusercontent.com/1113030/138301614-eec3b00f-ab15-4b42-8728-d26fe54440fb.png)

**Comment by bnjtgly** [comment]
I was able to make it work. Here's my nuxt.config.js. I fixed it by adding a refreshToken in my `/login` response. Now my problem is how to automatically refresh the token. Since my current code only trigger once. 
```
auth: {
    cookie: {
      options: {
        sameSite: 'lax',
      },
    },
    strategies: {
      local: {
        scheme: "refresh",
        token: {
          property: 'token',
          global: true
        },
        refreshToken: {
          property: "refreshToken",
          data: "refreshToken",
          tokenRequired: true
        },
        user: {
          property: 'user'
        },
        endpoints: {
          login: { url: 'login', method: 'post' },
          refresh: { url: 'refresh_token', method: 'post' },
          logout: { url: 'logout', method: 'delete' },
          user: { url: 'admin_api/v1/profile', method: 'get' }
        },
        redirect: {
          login: '/login',
          logout: '/login',
          callback: '/login',
          home: '/'
        }
      }
    },
    router: {
      middleware: ['auth']
    }
  }
```
